draft-packages/button/KaizenDraft/Button/components/GenericButton.tsx

```diff
@@ -149,23 +168,30 @@ const renderButton = (props: Props, ref: Ref<HTMLButtonElement>) => {
     disableTabFocusAndIUnderstandTheAccessibilityImplications,
     onFocus,
     onBlur,
+    working,
+    workingLabel,
+    ariaWorkingLabel,
+    iconButton,
     ...rest
   } = props
   const label = props.icon && props.iconButton ? props.label : undefined
   const customProps = getCustomProps(rest)
+  const btnWorkingLabel = getWorkingLabel(props)
+  const btnOnClick = !iconButton && working ? undefined : onClick
 
   return (
     <button
       id={id}
       disabled={disabled}
       className={buttonClass(props)}
-      onClick={onClick}
+      onClick={btnOnClick}
       onFocus={onFocus}
       onBlur={onBlur}
       onMouseDown={(e: any) => onMouseDown && onMouseDown(e)}
       type={type}
       title={label}
-      aria-label={label}
+      aria-label={working && btnWorkingLabel ? btnWorkingLabel : label}
+      aria-disabled={!!working}
       tabIndex={
         disableTabFocusAndIUnderstandTheAccessibilityImplications
```

Comment:
For documentation sake - I was going to say:

> I believe we'll also want to disable keyboard focus in the case that the button is `working` with `tabIndex={-1}`. You can just replace this line with a new var which is `const disableTabFocus =
    disableTabFocusAndIUnderstandTheAccessibilityImplications || working`.

On second thought, it's probably safer to not, even though it's not really interactable anymore.